db/write_thread.cc

```diff
@@ -781,13 +782,32 @@ void RequestQueue::CommitSequenceAwait(CommitRequest* req,
                                        std::atomic<uint64_t>* commit_sequence) {
   std::unique_lock<std::mutex> guard(commit_mu_);
   while (!requests_.empty() && requests_.front() != req && !req->committed) {
+    // When the subsequent commit finds that the front writer has not yet
+    // submitted, it will help the front writer to perform some tasks
+    auto front = requests_.front()->writer;
+    if (front->ConsumableOnOtherThreads()) {
```

Comment:
If I remember right, it's possible that during memtable write of current group, another write group can be formed to write WAL and added into the request queue. Previously it's not an issue because we do nothing unless the request is committed. But now you might consume those requests whose data isn't written into WAL yet.

Reply:
> it's possible that during memtable write of current group (A), another write group (B) can be formed to write WAL and added into the request queue. 

So A must have finished wrinting WAL right?

> But now you might consume those requests whose data isn't written into WAL yet.

I think this won't happen, as when A finished its memtable,  A returns directly, will not help B (only the following help the first)  no matter what stage B is in.

I think If there is no WAL allowed to write in parallel, there is no problem. Am I understanding some information wrong?


BTW,  I agree

> I suggest adding another atomic flag persisted to mark that a request is safe to be applied. It can solve the two issues in my comments.

Reply:
Then you need to check self.req.committed before entering this while loop. Because self.req might already be dequeued and it will loop to write group B.

Reply:
I can't figure out how it could happen.

Reply:
self.req.committed is already checked. My mistake.